vnext/Microsoft.ReactNative/Views/VirtualTextViewManager.cpp

```diff
@@ -20,28 +24,57 @@ using namespace xaml::Documents;
 namespace Microsoft::ReactNative {
 
 void VirtualTextShadowNode::AddView(ShadowNode &child, int64_t index) {
-  auto view = static_cast<ShadowNodeBase &>(child).GetView();
-
-  if (auto run = view.try_as<xaml::Documents::Run>()) {
-    transformableText.originalText = run.Text().c_str();
-    run.Text(transformableText.TransformText());
-  } else if (auto span = view.try_as<xaml::Documents::Span>()) {
+  auto &childNode = static_cast<ShadowNodeBase &>(child);
+  ApplyTextTransformToNode(childNode, textTransform, /* hasChanged = */false);
+  if (auto span = childNode.GetView().try_as<xaml::Documents::Span>()) {
     auto &childVTSN = static_cast<VirtualTextShadowNode &>(child);
-    auto transform = childVTSN.transformableText.textTransform;
-
-    for (const auto &i : span.Inlines()) {
-      if (auto run = i.try_as<xaml::Documents::Run>()) {
-        if (transform == TransformableText::TextTransform::Undefined) {
-          // project the parent transform onto the child
-          childVTSN.transformableText.textTransform = transformableText.textTransform;
-          run.Text(childVTSN.transformableText.TransformText());
-        }
+    m_highlightData.data.emplace_back(childVTSN.m_highlightData);
+  }
+  Super::AddView(child, index);
+}
+
+void VirtualTextShadowNode::ApplyTextTransform(TransformableText::TextTransform transform, bool hasChanged, bool inherited) {
+    if (!inherited || textTransform == TransformableText::TextTransform::Undefined) {
+    if (auto uiManager = GetNativeUIManager(GetViewManager()->GetReactContext()).lock()) {
+      for (auto childTag : m_children) {
+        const auto childNode = static_cast<ShadowNodeBase *>(uiManager->getHost()->FindShadowNodeForTag(childTag));
+        ApplyTextTransformToNode(*childNode, transform, hasChanged);
       }
     }
+  }
+}
 
-    m_highlightData.data.emplace_back(childVTSN.m_highlightData);
+void VirtualTextShadowNode::ApplyTextTransformToNode(
+    ShadowNodeBase &node,
+    TransformableText::TextTransform transform,
+    bool hasChanged) {
+  if (hasChanged ||
+      (transform != TransformableText::TextTransform::Undefined &&
+       transform != TransformableText::TextTransform::None)) {
+    const auto view = node.GetView();
+    const auto span = view.try_as<winrt::Span>();
+    const auto run = view.try_as<winrt::Run>();
+    if (span) {
+      auto &virtualTextChildNode = static_cast<VirtualTextShadowNode &>(node);
+      virtualTextChildNode.ApplyTextTransform(transform, hasChanged, /* inherited = */true);
+    } else if (run) {
+      auto &rawTextChildNode = static_cast<RawTextShadowNode &>(node);
+      auto originalText = rawTextChildNode.originalText;
+      if (originalText.size() != run.Text().size()) {
```

Comment:
can we get into a situation where these two have the same size but different content?
or vice-versa? w/surrogate pairs maybe?

Reply:
I *think* that raw text nodes are immutable. You cannot change the text prop on the raw text node, so once it's set, if the raw text value is changed, the entire Run will be replaced.

I am now questioning this, so I'll verify that raw text nodes are immutable and if they are leave a nice comment explaining why it's safe to use the length here :)

Reply:
Indeed this was a good catch. Raw text nodes are not immutable, we re-use the runs.

Reply:
Fixed.

Reply:
thanks for verifying :)